Userland/Applications/Maps/MapsWidget.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2023, Bastiaan van der Plaat <bastiaan.v.d.plaat@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibGUI/Painter.h>
+#include <LibGUI/Widget.h>
+#include <LibProtocol/RequestClient.h>
+
+class MapsWidget final : public GUI::Widget {
+    C_OBJECT(MapsWidget);
+
+public:
+    struct LatLng {
+        double latitude;
+        double longitude;
+    };
+
+    struct Options {
+        String tile_layer_url { "https://tile.openstreetmap.org/{}/{}/{}.png"_string };
+        LatLng center { 30, 0 };
+        int zoom { 3 };
+        bool attribution_enabled { true };
+        String attribution_text { "© OpenStreetMap contributors"_string };
+        URL attribution_url { "https://www.openstreetmap.org/copyright"sv };
+    };
+
+private:
+    MapsWidget(Options const&);
+
+    virtual void mousemove_event(GUI::MouseEvent&) override;
+
+    virtual void mousedown_event(GUI::MouseEvent&) override;
+
+    virtual void mouseup_event(GUI::MouseEvent&) override;
+
+    virtual void mousewheel_event(GUI::MouseEvent&) override;
+
+    virtual void paint_event(GUI::PaintEvent&) override;
+
+    struct Tile {
+        int x;
+        int y;
+        int zoom;
+        RefPtr<Gfx::Bitmap> image { nullptr };
+    };
+
+    Tile* get_tile(int x, int y);
+
+    void paint_tiles(GUI::Painter&);
+
+    void paint_attribution(GUI::Painter&);
+
+    int const TILE_SIZE = 256;
+    double const MAX_LATITUDE = 85.0511287798066;
+    size_t const MAX_TILES = 256;
+
+    NonnullRefPtr<Protocol::RequestClient> m_request_client;
+    String m_tile_layer_url;
+    LatLng m_center;
+    int m_zoom;
+    bool m_attribution_enabled;
+    String m_attribution_text;
+    URL m_attribution_url;
+    bool m_dragging { false };
+    int m_last_mouse_x { 0 };
+    int m_last_mouse_y { 0 };
+    Vector<Tile*> m_tiles;
```

Comment:
What's the reason for storing pointers here instead of the Tiles themselves?

Reply:
There was no reason really, I have reworked the code so that just `Tile`s are stored